omod/src/main/webapp/resources/scripts/metadata/manageRoles.js

```diff
@@ -0,0 +1,251 @@
+                templateUrl: "templates/list.page",
+                controller: "ManageRolesController"
+            })
+            .state("edit", {
+                url: "/edit/:roleUuid",
+                templateUrl: "templates/edit.page",
+                params: {
+                	roleUuid: null
+                },
+                resolve: {
+                    role: function($stateParams, Role) {
+                        if ($stateParams.roleUuid) {
+                            return Role.get({ uuid: $stateParams.roleUuid, v: "full", includeAll: true });
+                        }
+                        return {};
+                    }
+                },
+                controller: "EditRoleController"
+            });
+    }])
+
+    .controller("ManageRolesController", [ "$scope", "$state", "Role", "ngDialog", 
+        function($scope, $state, Role, ngDialog) {
+    		
+            function loadRoles() {
+                 Role.query({ v: "default", includeAll: true }).$promise.then(function(response) {
+                    // TODO handle multiple pages of results in a standard way
+                    $scope.roles = response.results;
+                }, function() {
+                    emr.errorMessage(emr.message("adminui.role.purge.success"));
+                })
+            }
+
+            
+            $scope.load = function() {
+            	loadRoles();
+            }
+
+            $scope.edit = function(role) {
+                $state.go("edit", { roleUuid: role.uuid });
+            }
+
+            $scope.purge = function(role) {
+                ngDialog.openConfirm({
+                    showClose: false,
+                    closeByEscape: true,
+                    closeByDocument: true,
+                    template: "templates/purgeRoleDialog.page",
+                    controller: function($scope) {
+                        $scope.role = role;
+                    }
+                	}).then(function() {
+	                    var toDelete = {
+	                    		uuid: role.uuid
+	                    }
+                    Role.delete(toDelete)
+                    .$promise.then(function() {
+                        emr.successMessage(emr.message("adminui.role.purge.success"));
+                        $state.reload(); 
+                    }, function() {
+                        emr.errorMessage(emr.message("adminui.role.purge.error"));
+                    });
+                });
+            }
+        }])
+
+    .controller("EditRoleController", [ "$scope", "$state", "Role", "role", "Privilege",
+        function($scope, $state, Role, role, Privilege) {
+
+            $scope.role = role;
+            
+            // test whether an object is contained within an array
+            function isInArray(myArray, myValue)
+            {
+            	var inArray = false;
+                if (myArray != null) {
+                	for (idx = 0; idx < myArray.length; idx++) {
```

Comment:
When i try to load some roles like Anonymous and System Developer, the script runs endlessly in the browser on this line and the inherited roles and privileges sections fail to load, can you look into why?